extensions/eclipse-che-theia-workspace/src/browser/che-quick-open-workspace.ts

```diff
@@ -0,0 +1,223 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import { inject, injectable } from 'inversify';
+import {
+    QuickOpenGroupItem,
+    QuickOpenItem,
+    QuickOpenMode,
+    QuickOpenModel
+} from '@theia/core/lib/common/quick-open-model';
+import { LabelProvider, QuickOpenService } from '@theia/core/lib/browser';
+import { CommandContribution, CommandRegistry } from '@theia/core/lib/common';
+import { WorkspaceCommands } from '@theia/workspace/lib/browser/workspace-commands';
+import { CheApiService } from '@eclipse-che/theia-plugin-ext/lib/common/che-protocol';
+import { che as cheApi } from '@eclipse-che/api';
+import { OauthUtils } from '@eclipse-che/theia-plugin-ext/lib/browser/oauth-utils';
+import { AbstractDialog } from '@theia/core/lib/browser/dialogs';
+import { MessageService } from '@theia/core/lib/common/message-service';
+import * as moment from 'moment';
+import { Message } from '@theia/core/lib/browser/widgets/index';
+import { Key } from '@theia/core/lib/browser/keyboard/keys';
+
+@injectable()
+export class QuickOpenCheWorkspace implements QuickOpenModel, CommandContribution {
+    protected items: QuickOpenGroupItem[];
+    protected currentWorkspace: cheApi.workspace.Workspace;
+
+    @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService;
+    @inject(CommandRegistry) protected readonly commandRegistry: CommandRegistry;
+    @inject(CheApiService) protected readonly cheApi: CheApiService;
+    @inject(OauthUtils) protected readonly oAuthUtils: OauthUtils;
+    @inject(LabelProvider) protected readonly labelProvider: LabelProvider;
+    @inject(MessageService) protected readonly messageService: MessageService;
+
+    registerCommands(commands: CommandRegistry): void {
+        // VS Code command id
+        commands.unregisterCommand('workbench.action.openRecent');
+
+        // Theia command id
+        commands.unregisterCommand('workspace:openRecent');
```

Comment:
I don't think we should unregister the command here: if an extension tries to execute `workbench.action.openRecent` it wants the original Theia command, not the Che version.
What we should do is unregister/reregister the menu items. There is an example of how to do that in https://github.com/eclipse/che-theia/pull/408/files#diff-698c505739271662ed1414a0a9c6314fR56

Reply:
>if an extension tries to execute `workbench.action.openRecent` it wants the original Theia command

Can you provide an example of existed extension, that might call this command?

I think, in a context of Che, it would be reasonable to operate with Che workspaces. I mean, in Theia (pure), workspace means something like content root, in our case it is a physical workspace (Che workspace). The same situation for `Close Workspace` command (which will be the next step), in pure Theia it simply closes the browser window, in our case it should stop the current workspace. @l0rd wdyt?

Reply:
The effect on the UI level of what I'm requesting is the exact same thing. Let's say we register our commands under the id `che.openRecentWorskpace` if we write some code that wants to call this command via the plugin API, we can just use that command. If we overwrite the command, plugins cannot call the original command anymore (even from a different menu), so offering "Open Recent Theia Workspace..." becomes impossible to implement. Also, we can't possibly know what the intention of third-party authors with respect to theia vs che workspaces is. My suspicion is they would want to call the original command, yours seems to be the opposite. The conservative move is to leave the original command in place and add a new one.

Reply:
You can see updates in latest commits. Is this okay for you?

Just note, if we remove default Theia's menu item (`Open Recent Workspace...`), the command is still registered and it is visible in command palette among with the registered Che command and menu item. Label for the Che menu item won't be changed. The label for Theia's one will be changed, but according to https://github.com/eclipse/che/issues/17106#issue-631935000 issue, not with this PR.